examples/tutorials/nb_python/asset_viewer.py

```diff
@@ -0,0 +1,972 @@
+        # set agent state to look at object
+        camera_position = agent.scene_node.translation
+        camera_look_at = objects[0].translation
+        for i in range(1, num_objs):
+            camera_look_at += objects[i].translation
+        camera_look_at /= len(objects)
+        agent.scene_node.rotation = mn.Quaternion.from_matrix(
+            mn.Matrix4.look_at(camera_position, camera_look_at, up_vec).rotation()  # up
+        )
+        if get_frames:
+            observations.append(sim.get_sensor_observations())
+
+    return observations
+
+
+# Set an object transform relative to the agent state
+def set_object_state_from_agent(
+    sim,
+    obj,
+    offset=np.array([0, 2.0, -1.5]),
+    orientation=mn.Quaternion(((0, 0, 0), 1)),
+):
+    agent_transform = sim.agents[0].scene_node.transformation_matrix()
+    ob_translation = agent_transform.transform_point(offset)
+    obj.translation = ob_translation
+    obj.rotation = orientation
+
+
+# %%
+# @title Define Visualization Utility Function { display-mode: "form" }
+# @markdown (double click to show code)
+# @markdown - display_sample
+
+# Change to do something like this maybe: https://stackoverflow.com/a/41432704
+def display_sample(
+    rgb_obs, semantic_obs=np.array([]), depth_obs=np.array([]), key_points=None
+):
+    from habitat_sim.utils.common import d3_40_colors_rgb
+
+    rgb_img = Image.fromarray(rgb_obs, mode="RGBA")
+
+    arr = [rgb_img]
+    titles = ["rgb"]
+    if semantic_obs.size != 0:
+        semantic_img = Image.new("P", (semantic_obs.shape[1], semantic_obs.shape[0]))
+        semantic_img.putpalette(d3_40_colors_rgb.flatten())
+        semantic_img.putdata((semantic_obs.flatten() % 40).astype(np.uint8))
+        semantic_img = semantic_img.convert("RGBA")
+        arr.append(semantic_img)
+        titles.append("semantic")
+
+    if depth_obs.size != 0:
+        depth_img = Image.fromarray((depth_obs / 10 * 255).astype(np.uint8), mode="L")
+        arr.append(depth_img)
+        titles.append("depth")
+
+    plt.figure(figsize=(12, 8))
+    for i, data in enumerate(arr):
+        ax = plt.subplot(1, 3, i + 1)
+        ax.axis("off")
+        ax.set_title(titles[i])
+        # plot points on images
+        if key_points is not None:
+            for point in key_points:
+                plt.plot(point[0], point[1], marker="o", markersize=10, alpha=0.8)
+        plt.imshow(data)
+
+    plt.show(block=False)
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--no-display", dest="display", action="store_false")
+    parser.add_argument("--no-make-video", dest="make_video", action="store_false")
+    parser.set_defaults(show_video=True, make_video=True)
+    args, _ = parser.parse_known_args()
+    show_video = args.display
+    display = args.display
+    make_video = args.make_video
+else:
+    show_video = False
+    make_video = False
+    display = False
```

Comment:
All the remaining code would still run though... This kinda defeats the point of `__main__`.